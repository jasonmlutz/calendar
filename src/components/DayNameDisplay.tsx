import React from "react";

type DayName = Readonly<
  | "Sunday"
  | "Monday"
  | "Tuesday"
  | "Wednesday"
  | "Thursday"
  | "Friday"
  | "Saturday"
>;

const dayNames: DayName[] = [
  "Sunday",
  "Monday",
  "Tuesday",
  "Wednesday",
  "Thursday",
  "Friday",
  "Saturday",
];

const renderedDayNames: JSX.Element[] = dayNames.map((name, index) => (
  <li key={index}>{name}</li>
));

function DayNameDisplay() {
  return <ul className="flex flex-row justify-between">{renderedDayNames}</ul>;
}

export default DayNameDisplay;
