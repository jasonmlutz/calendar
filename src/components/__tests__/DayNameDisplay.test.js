import renderer from "react-test-renderer";
import { unmountComponentAtNode } from "react-dom";
import { render, screen } from "@testing-library/react";
import { act } from "react-dom/test-utils";

import DayNameDisplay from "../DayNameDisplay";

it("renders correctly", () => {
  const tree = renderer.create(<DayNameDisplay />).toJSON();
  expect(tree).toMatchSnapshot();
});

let container = null;
beforeEach(() => {
  // setup a DOM element as a render target
  container = document.createElement("div");
  document.body.appendChild(container);
});

afterEach(() => {
  // cleanup on exiting
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

it("contains the days of the week", () => {
  act(() => {
    render(<DayNameDisplay />, container);
  });
  expect(screen.getByRole("list").textContent).toBe("");
});
