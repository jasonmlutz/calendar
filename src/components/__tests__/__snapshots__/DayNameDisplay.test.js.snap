// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<ul
  className="flex flex-row justify-between"
>
  <li>
    Sunday
  </li>
  <li>
    Monday
  </li>
  <li>
    Tuesday
  </li>
  <li>
    Wednesday
  </li>
  <li>
    Thursday
  </li>
  <li>
    Friday
  </li>
  <li>
    Saturday
  </li>
</ul>
`;
